src/pocketmine/Player.php

```diff
@@ -2136,8 +2136,10 @@ public function handleDataPacket(DataPacket $packet){
 						break;
 					case PlayerActionPacket::ACTION_RELEASE_ITEM:
 						if($this->startAction > -1 and $this->getDataFlag(self::DATA_FLAGS, self::DATA_FLAG_ACTION)){
-							if($this->inventory->getItemInHand()->getId() === Item::BOW){
-								$bow = $this->inventory->getItemInHand();
+							if($this->inventory->getItemInHand()->getId() === Item::BOW
+                                and !$this->isSpectator()){
```

Comment:
I wasn't quite sure whether to put the `and !$this->isSpectator()` onto a new line or not because it would have made the header of the if statement longer, but I also realize that some people don't like long headers, so I put it on the next line and then inserted another line after for readability's sake.